deriving Container produce warning "Redundant constraint"

Using ghc-8.6.3.

```Haskell
{-# LANGUAGE GeneralizedNewtypeDeriving, UndecidableInstances, FlexibleContexts, GADTs #-}
module Test where
import Universum

newtype Test = Test [Int]
    deriving (Container)
```

Build with ``stack ghc -- -fwarn-redundant-constraints %``, result:

```
test.hs:6:15: warning: [-Wredundant-constraints]
    • Redundant constraint: Eq (Element Test)
    • In an expression type signature:
           Eq (Element Test) => Element Test -> Test -> Bool
      In the expression:
          GHC.Prim.coerce
            @(Element ([] Int) -> [] Int -> Bool)
            @(Element Test -> Test -> Bool)
            Universum.elem ::
            Eq (Element Test) => Element Test -> Test -> Bool
      In an equation for ‘Universum.elem’:
          Universum.elem
            = GHC.Prim.coerce
                @(Element ([] Int) -> [] Int -> Bool)
                @(Element Test -> Test -> Bool)
                Universum.elem ::
                Eq (Element Test) => Element Test -> Test -> Bool
      When typechecking the code for ‘Universum.elem’
        in a derived instance for ‘Container Test’:
        To see the code I am typechecking, use -ddump-deriv
  |
6 |     deriving (Container)
  |               ^^^^^^^^^

```

--- a/test/Test/Universum/Issue208.hs
+++ b/test/Test/Universum/Issue208.hs
@@ -1,0 +1,23 @@
+{-# LANGUAGE
+    GeneralizedNewtypeDeriving
+  , UndecidableInstances
+  , FlexibleContexts
+  , GADTs
+  , DerivingStrategies
+  , CPP
+#-}
+module Test.Universum.Issue208
+  () where
+
+#if __GLASGOW_HASKELL__ >= 900
+
+import Universum (Container)
+
+-- In ghc-8.6.3 this code will produce a @redundant constraint@ warning.
+-- In ghc-9.0.2 and newer no warnings would be produced.
+-- Issue #208: https://github.com/serokell/universum/issues/208
+
+newtype Test = Test [Int]
+    deriving newtype (Container)
+
+#endif

--- a/universum.cabal
+++ b/universum.cabal
@@ -121,7 +121,9 @@ test-suite universum-test
   hs-source-dirs:      test
   main-is:             Spec.hs
 
-  other-modules:       Test.Universum.Property
+  other-modules:
+                       Test.Universum.Issue208
+                       Test.Universum.Property
 
   build-depends:       universum
                      , bytestring

